src/Sema.zig

```diff
@@ -5773,10 +5774,17 @@ fn zirSatArithmetic(
     defer tracy.end();
 
     const extra = sema.code.extraData(Zir.Inst.SaturatingArithmetic, extended.operand).data;
-    const src: LazySrcLoc = .{ .node_offset = extra.node };
-    return sema.mod.fail(&block.base, src, "TODO implement Sema.zirSatArithmetic", .{});
+    sema.src = .{ .node_offset_bin_op = extra.node };
+    const lhs_src: LazySrcLoc = .{ .node_offset_bin_lhs = extra.node };
+    const rhs_src: LazySrcLoc = .{ .node_offset_bin_rhs = extra.node };
+    const lhs = sema.resolveInst(extra.lhs);
+    const rhs = sema.resolveInst(extra.rhs);
+
+    return sema.analyzeArithmetic(block, .extended, lhs, rhs, sema.src, lhs_src, rhs_src, extended);
 }
 
+// TODO: audit - not sure if its a good idea to reuse this, adding `opt_extended` param
+// FIXME: somehow, rhs of <<| is required to be Log2T. this should accept T
 fn analyzeArithmetic(
     sema: *Sema,
     block: *Scope.Block,
```

Comment:
I'm reusing analyzeArithmetic() by adding an `opt_extended` param here.  Is this a bad idea?

Also, I can't figure out where the check that rhs of <<| is a Log2T is happening (and how to disable it).  It needs to accept T, but currently doesn't.

Reply:
> Why do you think requiring an @intcast for the left-shift is a bad idea?

@matu3ba I was just trying to match the behavior of @shlWithSaturation which doesn't require Log2T, but accepts T.

Reply:
also, this is how the [llvm intrinsics](https://llvm.org/docs/LangRef.html#llvm-sshl-sat-intrinsics) are defined